odoo/orm/environments.py

```diff
@@ -865,7 +855,25 @@ def clear_dirty_field(self, field: Field) -> Collection[IdType]:
         return self.transaction.dirty.pop(field, ())
 
     def invalidate(self, spec: Collection[tuple[Field, Collection[IdType] | None]] | None = None) -> None:
-        return self.transaction.invalidate(spec)
+        """ Invalidate the cache, partially or totally depending on ``spec``.
```

Comment:
All logic should be removed from `Cache` object. Can you move it back to transaction object?